RadixWallet/Clients/TokenPriceClient/TokenPriceClient+Live.swift

```diff
@@ -45,10 +42,12 @@ extension TokenPricesClient {
 extension TokenPricesClient.TokenPrices {
 	fileprivate init(_ tokenPricesResponse: TokensPriceResponse) {
 		let formatter = NumberFormatter()
+		formatter.locale = Locale(identifier: "en_US_POSIX") // Just ignore the users locale
 		formatter.numberStyle = .decimal
 		formatter.maximumFractionDigits = Int(RETDecimal.maxDivisibility)
 		formatter.roundingMode = .down
-		formatter.decimalSeparator = "." // Enfore dot notation for RETDecimal
+		formatter.decimalSeparator = "." // Enforce dot notation for RETDecimal
+		formatter.usesGroupingSeparator = false // No grouping separator for RETDecimal
```

Comment:
Not having this caused the prices > 1000 to be formatted with a thousand separator which then couldn't be converted to RETDecimal.

Reply:
there is another initialiser than `value: String`, there is `formatted: String` as well. which should be able to accept strings with a thousands separator.

Reply:
It looks like we are rounding the Double that represents the price, by first converting the price Double to a string (using a fixed decimal separator) and then converting it back to RETDecimal? 

1. I presume that `RETDecimal(value: String)` is the version that takes a machine readable string, so that it ignores the user preference on decimal separator, because otherwise this wouldn't work, but even so, I don't like this, it doesn't seem nice or necessary to convert the price to a Double, then a String and the RETDecimal, just in order to round it.

2. If we do want to round it here, can't we convert it (back) to RETDecimal and just round that, without going by way of String?

3. Why do we want `price` to be a Double at all? This operation here would be even simpler if it was kept a RETDecimal until it's to be displayed.

4. Is the behaviour actually correct? What is the purpose of clamping the number of decimals in the **price** according to the divisibility, isn't the price here given in a fiat currency?

Reply:
>there is another initialiser than value: String, there is formatted: String as well. which should be able to accept strings with a thousands separator.

Presumably (hopefully), the latter respects the user's locale, and the former does not, and in this case we don't want to respect the locale, since we hard code the decimal separator.

Reply:
@kugel3 note this whole shenanigans is not about rounding, it is about truncating to RETDecimal max divisibility, otherwise we cannot initialize the RETDecimal with a double that has more decimals than 18.

The price is double as it comes from the TokenPrice service, here we do immediately convert it to RETDecimal.

Reply:
Alright, then I understand your other comment about an initialiser for RETDecimal that takes a Double with arbitrary number of decimals, that would be highly preferable. Since NumberFormatter is pretty expensive to initialise, we probably don't want to make that initialiser in Swift.

But while we have this workaround, I think we should be more explicit about what is going on, so that nobody accidentally changes to using the other string based initialiser for example. Because what I wrote before is correct, right? The `RETDecimal(formatted: String)` could not actually be used here?

Perhaps change `// Enforce dot notation for RETDecimal` to something like `// Use a period as decimal separator, for the sake of RETDecimal(formatted: String)`.

Reply:
Yes `RETDecimal.init(formattedStrinf:locale)` is respects locale